Address RuboCop violations that couldn't be automatically fixed

mokagio opened the pull request:
Continues the from #18839, addressing the remaining violation that couldn't be automatically fixed.

To test... read the code?

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

mokagio committed:
Ignore a `Metrics/AbcSize` RuboCop violation in `Rakefile`

I couldn't see any straightforward way to address the violation.

Given `Rakefile` is not a core component of the codebase, but only a
runner used for a limited set of tasks, I think it's acceptable to just
ignore the violation and move on.

mokagio committed:
Group all `Rakefile` `require` together

mokagio committed:
Replace get Ruby version method with a constant in `Rakefile`

RuboCop highlighted the method because it started with `get_`. I thought
it made more sense to remove an unnecessary method rather than rename
it.

mokagio committed:
Add `Rakefile` to the `Exclude` list of some RuboCop rules

mokagio committed:
Remove unnecessary Travis CI check from `Rakefile`

We haven't been using Travis for a long time.

mokagio committed:
Drop `get_` prefix from `get_xcode_version` in `Rakefile`

mokagio committed:
Replace `get_ci_xcode_version` method with constant in `Rakefile`

mokagio committed:
Drop `get_` prefix from `client_id` and `client_secret` in `Rakefile`

mokagio committed:
Simplify a nested `if` in `Rakefile`

mokagio committed:
Replace a `rescue Exception` with `rescue StandardError`

This is a RuboCop recommendation.

See
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/RescueException

mokagio committed:
Don't make a frozen string a CONSTANT when in a block in `Rakefile`

See
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/ConstantDefinitionInBlock

> Do not define constants within a block, since the block's scope does
> not isolate or namespace the constant in any way.

mokagio committed:
Remove empty block syntax in `target 'Jetpack'` in `Podfile`

mokagio committed:
Address a `Sytle/CombinableLoops` violation in `Podfile`

mokagio committed:
Opt-out from cops that are impractical for `Podfile` in `.rubocop.yml`

mokagio commented:
Bumping to the 20.2 milestone because I'm code freezing 20.1 and this hasn't been reviewed yet.

AliSoftware commented on Podfile:
```diff
@@ -467,11 +466,20 @@ post_install do |installer|
       config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)', 'ALPHA=1'] if (config.name == 'Release-Alpha') || (config.name == 'Release-Internal')
     end
   end
+  # rubocop:enable Sytle/CombinableLoops
```
See https://github.com/wordpress-mobile/WordPress-iOS/pull/18847#discussion_r897909868

AliSoftware reviewed (commented):
I left most of my feedback about the changes seen here in [this other PR 18847](https://github.com/wordpress-mobile/WordPress-iOS/pull/18847) which does the same and more.

mokagio committed:
Fix typo in `Podfile` (uless -> unless)

mokagio committed:
Add `rstrip` to a `File.read` call in `Rakefile`

"Just to be on the safe side in case someone adds a CRLF at the end of
this file in the future."

See https://github.com/wordpress-mobile/WordPress-iOS/pull/18847#discussion_r897913331

mokagio committed:
Simplify a loop in `Podfile`

See https://github.com/wordpress-mobile/WordPress-iOS/pull/18847#discussion_r897909868

mokagio commented:
@AliSoftware thank you for the suggestions, can you have another look? Thanks!

I regret not realizing that I messed up the PRs stacking between this and https://github.com/wordpress-mobile/WordPress-iOS/pull/18847. #18847 should only have had the Danger-related commits 😞 

Note that I disabled auto-merge intentionally because I want to rebase to get rid of the commit on this branch before merging on `trunk`.

AliSoftware commented on fastlane/lanes/localization.rb:
```diff
@@ -327,6 +330,7 @@ def download_localized_app_store_metadata(glotpress_project_url:, locales:, meta
           + "Delete the `#{en_file_path}` file, ensure the `default/` one has the expected original copy, and try again.")
       end
     end
+    # rubocop:enable Metrics/AbcSize
```
This seems to be at the wrong place. The corresponding `rubocop:disable` on line 302 is at the same level as the function declaration (indentation level 1), while this balancing one is inside the function body (indentation level 2).

We should either move the `disable` on line 305 inside the function body, or move the `enable` on line 353 after the `end` of the function body.

mokagio replied:
Good catch, thanks! Clearly went for the wrong `end` 🤦‍♂️ Will move to the end of the method.

AliSoftware reviewed (approved):
Left only one nitpick about `rubocop:enable` location, otherwise LGTM.

Approving to unblock you, as you disabled auto-merge and I trust you'll fix this nitpick before your rebase and merge 🙂

mokagio committed:
Correctly fence an entire method with a RuboCop disable comment

Previously, the enable was in the method body, which was confusing.

See
https://github.com/wordpress-mobile/WordPress-iOS/pull/18845#discussion_r908478959
